test/unit/org/apache/cassandra/index/sai/cql/VectorUpdateDeleteTest.java

```diff
@@ -489,6 +488,8 @@ public void shadowedPrimaryKeyWithSharedVector()
         // flush a sstable with one vector that is shared by two rows
         execute("INSERT INTO %s (pk, str_val, val) VALUES (0, 'A', [1.0, 2.0, 3.0])");
         execute("INSERT INTO %s (pk, str_val, val) VALUES (2, 'B', [1.0, 2.0, 3.0])");
+        // Add another row to ensure we get the correct ordinal from disk
```

Comment:
i'm slightly confused, if the test doesn't pass w/o this then we still have a bug, no?

Reply:
I added this row to exercise the delegated portion of `AugmentedRowIdsView#getSegmentRowIdsMatching`. The test passes with and without the changes to the test.